components/bl-barcode-component/src/index.js

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useMemo, useRef } from 'react';
+
+import './lib/barcode.min';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const FontOptions = {
+  UNSET      : 'unset',
+  BOLD       : 'bold',
+  ITALIC     : 'italic',
+  BOLD_ITALIC: 'boldItalic',
+};
+
+const FontOptionsMap = {
+  [FontOptions.UNSET]      : '',
+  [FontOptions.BOLD]       : 'bold',
+  [FontOptions.ITALIC]     : 'italic',
+  [FontOptions.BOLD_ITALIC]: 'bold italic',
+};
+
+export default function BarcodeComponent({ component, eventHandlers, elRef }) {
+  const { classList, display, style } = component;
+  const { onClick, onMouseOver, onMouseOut } = eventHandlers;
+
+  const barcodeRef = useRef(null);
+
+  useBarcodeLibrary(component, barcodeRef);
+
+  const styles = {
+    display: !display && 'none',
```

Comment:
Why we use `display:none` but not `return null`?

Reply:
Due to the fact that we pass the ref to the library to generate a barcode, and if it does not render, the page load will break due to a library issue.

To avoid this we can add a condition before calling barcode generator and add "display" property to dependencies array.
```
  useEffect(() => {
    if (!barcodeRef.current) {
      return;
    }

    if (value) {
      JsBarcode(barcodeRef.current, value, { ...options, valid: onValidate });
    } else {
      barcodeRef.current.removeAttribute('src');
    }
  }, [display, value]);
```